src/clj/puppetlabs/services/jruby/jruby_service.clj

```diff
@@ -20,12 +21,15 @@
                            [:ShutdownService shutdown-on-error]]
   (init
     [this context]
-    (let [config (core/initialize-config (get-config))
+    (let [initial-config (get-config)
           service-id (tk-services/service-id this)
-          agent-shutdown-fn (partial shutdown-on-error service-id)]
+          agent-shutdown-fn (partial shutdown-on-error service-id)
+          config (core/initialize-config (assoc-in initial-config
+                                                   [:jruby :lifecycle :shutdown-on-error]
+                                                   agent-shutdown-fn))]
       (core/verify-config-found! config)
```

Comment:
Yeah this seems like pointless validation, and if we wanted that kind of validation is should be in the return schema of `initialize-config` or somewhere